### PR TITLE
do not attempt to call memcpy with NULL as 2nd argument

### PR DESCRIPTION
**Pull request by pbanicev**
Found with valgrind when running vector_boost_engine.

### CODE REVIEW

File: engine/src/ExternalAgentInterface/vb_engine_EA_interface.c [hunk 1]
```diff
@@ -551,7 +551,8 @@ t_VB_engineErrorCode VbEngineEAInterfaceSendFrame(
 
   if (result == VB_ENGINE_ERROR_NONE)
   {
-    memcpy((char *)msg->eaPayload.msg, payload, payloadLength);
+    if (payload && payloadLength > 0)
```
Comment (vgraumxl) [inline engine/src/ExternalAgentInterface/vb_engine_EA_interface.c:554]: Please, add parenthesis for each check.
Send message to VB only if conditions are met:

```suggestion
    if ((payload != NULL) && (payloadLength > 0))
    {
        memcpy((char *)msg->eaPayload.msg, payload, payloadLength);
        
         // Send frame
        VbEAMsgSend(&thisDriver->vbEAConnDesc, msg);
    }
```

**Review by vgraumxl — CHANGES_REQUESTED** [review]

**Comment by pbanicev** [comment]
I think I tried that approach (not sending anything) first and it caused regression in functionality.

File: engine/src/ExternalAgentInterface/vb_engine_EA_interface.c (hunk 1 above)
Comment (pbanicev) [reply engine/src/ExternalAgentInterface/vb_engine_EA_interface.c:554]: I think I tried that approach (not sending anything) first and it caused regression in functionality.

Comment (vgraumxl) [reply engine/src/ExternalAgentInterface/vb_engine_EA_interface.c:554]: oh, maybe I missed something, ok. 
So, can you please just change the "if" to check explicitly with NULL and using () and {}?

Comment (pbanicev) [reply engine/src/ExternalAgentInterface/vb_engine_EA_interface.c:554]: yes, will do.

**Pull request merged**